### ISSUE TITLE
encoding changes

### ISSUE THREAD

**Issue by noSoup84** [ashkulz/NppFTP#27]
I can create a file using UTF-8 encoding in Npp, and place it on my web server using a standalone FTP client. From here, when I download the file using NppFTP, the encoding has changed to ANSI, and the EOF also change from "UNIX" (default in my Npp configuration), to "Dos/Windows"

This also happens if I create the file on the server using NppFTP and download it using NppFTP.

**Comment by Laubeee** [comment]
Yeah, when I create a file with NppFTP, it is always ANSI encoded... it would be lovely if it would take over the Npp-Configuration which is usually UTF8-without-BOM.

**Comment by mfinky** [comment]
To: Laubeee

Hi.
Hoping this helps: when i open a new file it DOES show "UTF-8 w/o BOM"
i am using v6.7.5 on windows 8.1

Cheers.
marcelo finkielsztein

________________________________________

**Comment by Laubeee** [comment]
mfinky: you create the file with NppFTP? I am using the very same version (s)

**Comment by noSoup84** [comment]
OS: Windows 8.1
NPP: v6.7.7 (newest)
NppFTP: v0.26.2 unicode (newest)
NPP config: 
-default new file encoding: UTF-8 without BOM
-default line ending (EOL): UNIX/OSX

When I create a file in Npp, it gets the right Encoding and line ending(EOL), but when I create a file on a remote server using NPP, it shows up as Dos/Windows line ending and ANSI encoding,

From here I convert the file to correct encoding and line ending before I upload it.

On re-download from server it is back to ANSI and Dos/Windows EOL.

**Comment by mfinky** [comment]
Ohh,,,, sh@!%t!  I am sorry.
I should have clarified that.
I actually DO NOT create it on the server side with nppftp.  Sorry!.  I create the file on the “left” side, using regular npp, like any other “new” file.
And first time I save it, I save it on the specific ‘cache’ folder.
In my case:  C:\Users\marcelo\AppData\Roaming\Notepad++\plugins\Config\NppFTP\Cache\somewebsite@ftp.somewebsite.com\public_html

I noticed that when I save *there*, the file gets uploaded to the server.

Note: usually I first *download* some file from the website, into the cache.  Then, when I press “New”, npp suggests itself using the same folder as the already opened file, so it points already to the cache folder I wanted to use.
(as an alternative, you could right click on the tab of the opened file and click on {Current Dir Path to clipboard}.  I am SURE you already knew that.

So … hope this helps, at least as workaround?

I am not at home right now, I am writing this email from the office.  When I get back home I will try creating a new file on the ‘right’ side of the screen, on the server, and will let you know what I see.
Again, sorry I confused you.

Cheers.
Marcelo

**Comment by mfinky** [comment]
Hi Everyone.  Sorry for the confusion.

I should have clarified that.
I actually DO NOT create it on the server side with nppftp.  Sorry!.  I create the file on the “left” side, using regular npp, like any other “new” file.
And first time I save it, I save it on the specific ‘cache’ folder.
In my case:  C:\Users\marcelo\AppData\Roaming\Notepad++\plugins\Config\NppFTP\Cache\somewebsite@ftp.somewebsite.com\public_html

I noticed that when I save *there*, the file gets uploaded to the server.

Note: usually I first *download* some file from the website, into the cache.  Then, when I press “New”, npp suggests itself using the same folder as the already opened file, so it points already to the cache folder I wanted to use.
(as an alternative, you could right click on the tab of the opened file and click on {Current Dir Path to clipboard}.  I am SURE you already knew that.

So … hope this helps, at least as workaround?

I am not at home right now, I am writing this email from the office.  When I get back home I will try creating a new file on the ‘right’ side of the screen, on the server, and will let you know what I see.
Again, sorry I confused you.

Cheers.
Marcelo

**Comment by sebcar** [comment]
Same problem here.
Windows 7 64 bits.
NPP: 6.8.3
NppFTP: 0.26.3 Unicode

**Comment by andregarkin** [comment]
I have the same problem: 
when I download the file using NppFTP, the encoding has changed from UTF-8 to ANSI, and the EOF also change from "UNIX" (default in my Npp configuration), to "Dos/Windows"

Windows 7 64 bits.
NPP: 6.9
NppFTP: 0.26.3 (but I don't Unicode or not)

**Comment by jomofcw** [comment]
Hello,

Same issue there.
Please, fix it.

IMO NppFTP has to refer to the default encoding for new document when opening a document without being able to determine the encoding.

**Comment by msc1** [comment]
Hello, bumping nearly 2 years old thread. Will there be any fixes? Otherwise great plugin, please fix :)

**Comment by Turbine1991** [comment]
Also having this issue. The program which loads the lua file errors when saved through NppFTP. It works okay if I save it locally and then upload it via Filezilla.
